addon/services/engine-router-service.js

```diff
@@ -0,0 +1,118 @@
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import { action, computed, get } from '@ember/object';
+import { reads } from '@ember/object/computed';
+import { getOwner } from '@ember/application';
+import Evented from '@ember/object/evented';
+import { namespaceEngineRouteName } from '../utils/namespace-engine-route-name';
+import { getRootOwner } from '../utils/owner';
+
+export default class EngineRouterService extends Service.extend(Evented) {
+  init() {
+    super.init();
+
+    this._externalRoutes = getOwner(this)._externalRoutes;
+    this._mountPoint = getOwner(this).mountPoint;
+    this.rootApplication = getRootOwner(this);
+
+    this.externalRouter.on('routeWillChange', this.onRouteWillChange);
+    this.externalRouter.on('routeDidChange', this.onRouteDidChange);
+  }
+
+  destroy() {
+    this.externalRouter.off('routeWillChange', this.onRouteWillChange);
+    this.externalRouter.off('routeDidChange', this.onRouteDidChange);
+
+    super.destroy();
+  }
+
+  @action
+  onRouteWillChange(...args) {
+    this.trigger('routeWillChange', ...args);
+  }
+
+  @action
+  onRouteDidChange(...args) {
+    this.trigger('routeDidChange', ...args);
+  }
+
+  @reads('externalRouter.rootURL') rootURL;
+
+  @reads('externalRouter.currentURL') currentURL;
+
+  @computed('externalRouter.currentRouteName')
+  get currentRouteName() {
+    if (get(this, 'externalRouter').currentRouteName === this._mountPoint) {
+      return 'application';
+    }
+    return get(this, 'externalRouter').currentRouteName.slice(this._mountPoint.length + 1);
+  }
+
+  get externalRouter() {
+    return this.rootApplication.lookup('service:router');
+  }
```

Comment:
I don't think this is solved like suggested in https://github.com/ember-engines/ember-engines/pull/669#discussion_r309728284 yet.

Reply:
I think @rwjblue was referring to this code specifically:

https://github.com/emberjs/ember.js/blob/08e4da20ce49c5cd58fd9c8113624409fa7bd4ec/packages/%40ember/engine/index.js#L492-L529

Reply:
That's what I did based on `cloneParentDependencies` that has `router:main` from host app. Then I resolved using `router:main` there's only one instance in the ember app and it doesn't exist on engines. I consulted @rwjblue before implement it. 

https://github.com/ember-engines/ember-engines/pull/669/files#diff-ef6a68c610c2dff0c5f5d27e570c8f73